api/client/container/stats.go

```diff
@@ -192,12 +192,23 @@ func runStats(dockerCli *client.DockerCli, opts *statsOptions) error {
 
 	for range time.Tick(500 * time.Millisecond) {
 		printHeader()
+		toRemove := []int{}
 		cStats.mu.Lock()
-		for _, s := range cStats.cs {
+		for i, s := range cStats.cs {
 			if err := s.Display(w); err != nil && !opts.noStream {
 				logrus.Debugf("stats: got error for %s: %v", s.Name, err)
+				if err == io.EOF {
+					toRemove = append(toRemove, i)
+				}
 			}
 		}
+		for j := len(toRemove) - 1; j >= 0; j-- {
```

Comment:
Yes. I tried to use that.
But I found if you `docker stats` with partial container id or container name, the name stored in `cStats.cs` will be the the args which could be not easy to compare to the event sent by server.

for example:
```
$docker stats 5es test  #5es is container id, test is container name
```
We can only get the container id from event, so we cannot simply remove container with name.
And for container id we don't know the length we should slice.

Please correct me if I was wrong.
And if the problem need to be solved, I would like to do that.